### PR TITLE
Fix `ListBox` selection interactions

### LINKED ISSUE #3146: ListBox inside ScrollViewer prevent touch scrolling
If a ListBox is placed inside a ScrollViewer it steals the pointer pressed input (for item selection) preventing scrolling using touch when the touch down is inside the ListBox.

```
  <ScrollViewer>
    <StackPanel>
      <other controls.../>
      <ListBox Items="{Binding Items}"/>
    </StackPanel>
  </ScrollViewer>
```

The ListBox OnPointerPressed method sets e.Handled=true which prevent the PointerPressed event bubbling up to the ScrollViewer.

I would like to be able to both select items in the ListBox and scroll the StackPanel content using touch.

### LINKED ISSUE #9707: Improving pointer behavior in SelectingItemsControls
Currently, Avalonia's pointer behavior in SelectingItemsControls differs from other UI frameworks. In Avalonia, selecting an item (in a ComboBox/ListBox/TabControl/Menu) is all done on PointerPressed. In addition we also allow selection using the right mouse button. In other frameworks (all of Windows (Win32 and Xaml), Qt, gnome on linux for alternate OS comparison) invokes or selects with only the left mouse button and on pointer up. The only exception to this rule is top level menu items in a MenuBar (Windows HMENU or Qt menubar, for example) where the dropdown opens on pointer down. The menu items in the drop down though still invoke on pointer up.

We also don't clear the ":pressed" state if the pointer is pressed but dragged outside the bounds of the item.

Advantages to using only left button and pointer up:
- You can cancel an accidental pointer/mouse press - just drag the mouse outside the bounds of the item and release and the capture releases, but the item itself doesn't select/invoke
- Not handling right mouse means the item won't get selected if a ContextMenu/Flyout is opened on the item. (Add one to a ListBoxItem and you'll see the ListBoxItem get selected before the menu opens)
- The current behavior renders the `:pressed` pseudoclass less important since the pointer-triggered action has already happened, but the pressed visual state indicates you're about to do something to this item

I don't think this is too big of a change to actually implement. It will break the current behavior, but it will align it with other UI frameworks and what I think others would expect. Two things to consider with the change though: pointer up should only invoke/select the captured control (moving to another item and releasing should not select that item). And, `MenuItem` will need to adapt whether its toplevel or actually in a menu - this might be the more complicated bit?

With the other items control related changes going on, this might be a good time to adjust this - though it is of different scope to that PR.

### LINKED ISSUE #11079: Scrolling in a ListBox changes the SelectedItem.
**Describe the bug**
Scrolling in a large items ListBox control in a touchscreen (Avalonia XPlat Android Device) change the SelectedItem of the ListBox.

The SelectedItem of the ListBox should be changed if the finger touch the screen in a single tap, but now if you use your finger to scrolldown the listbox, the selecteditem is changed and this is wrong. 

**To Reproduce**
 - Create Avalonia XPlatform app with a ListBox of items, the items count should be engoht to create a scroll in the ListView.
 - Create a CompiledBinding TwoWays in the SelectedItem and navigate to a detail view on the setter method.

**Expected behavior**
Yo should navigate to any item at the bottom of the scroll.

**Desktop (please complete the following information):**
Only happens on touch screen devices.
Avalonia version: 11.0.0-preview5
Development enviroment: MacBook Pro M2.
Deployment device: Xiaomi Pad 5

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

Improves `ListBox` selection interactions:

- Selection is done on pointer release instead of pointer press when using touch. (#9707, #11079) 
- Selection no longer follows focus
- Ctrl+arrow keys move focus but not selection
- Selection outside visible viewport is correctly handled (arrow keys move selection from focused item and scroll it into view)

Implementation details:

- Pointer selection logic is moved from `ListBox` to `ListBoxItem`
- "Select All" and "Enter"/"Space" key selection is moved from `SelectingItemsControl` to `ListBox` (the docs for `SelectingItemsControl` say "does not react to user input")
- Added `ItemsControl.ItemsControlFromItemContaner` as in WPF

## Fixed issues

Fixes #3146
Fixes #9707
Fixes #11079

### CODE REVIEW

File: src/Avalonia.Controls/ListBoxItem.cs [hunk 1]
```diff
@@ -40,5 +45,67 @@ protected override AutomationPeer OnCreateAutomationPeer()
...
+                    }
+                    else
+                    {
+                        // Otherwise perform the selection when the pointer is released as to not
+                        // interfere with gestures.
+                        _pointerDownPoint = p.Position;
+
+                        // Ideally we'd set handled here, but that would prevent the scroll gesture
+                        // recognizer from working.
+                        ////e.Handled = true;
```
Comment (grokys) [inline src/Avalonia.Controls/ListBoxItem.cs:78]: This line should be uncommented when we fix scroll gestures to work with handled events.

**Review by emmauss — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035249-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Controls/VirtualizingStackPanel.cs [hunk 2]
```diff
@@ -265,12 +265,14 @@ protected override void OnItemsChanged(IReadOnlyList<object?> items, NotifyColle
         protected override IInputElement? GetControl(NavigationDirection direction, IInputElement? from, bool wrap)
         {
             var count = Items.Count;
+            var fromControl = from as Control;
 
-            if (count == 0 || from is not Control fromControl)
+            if (count == 0 || 
+                fromControl is null && direction is not NavigationDirection.First or NavigationDirection.Last)
```
Comment (jnyrup) [inline src/Avalonia.Controls/VirtualizingStackPanel.cs:271]: Should this have been `direction is not (NavigationDirection.First or NavigationDirection.Last)`?

Comment (timunie) [reply src/Avalonia.Controls/VirtualizingStackPanel.cs:271]: @jnyrup answering closed / merged PRs isn't really a good idea. Make it a new discussion or issue if you have an actual issue. Or open a PR to add the proposed fix if you're certain it is an issue.

**Pull request merged**